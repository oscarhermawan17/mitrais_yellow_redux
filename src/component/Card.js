import React from 'react';

function date_fe(value){
    return `${new Date(value).getDate()}/${new Date(value).getMonth() + 1}/${new Date(value).getFullYear()} ${new Date(value).getHours()}:${new Date(value).getMinutes()}`
}

class Card extends React.Component{
    render(){
        return(
            this.props.todos.map(todo => 
                <div className="todo_content" key={todo.id}>
                    <div>{todo.description}</div>
                    <div>{date_fe(todo.deadline)}</div>
                    <div className="manage_btn">
                        <button onClick={() => this.props.updateSingleTodo(todo)} className="btn gold">Edit</button> &nbsp;
                        <button onClick={() => this.props.deleteSingleTodo(todo.id)} className="btn merah">Delete</button>
                    </div>
                </div>
        ))
    }
        


}
    

export default Card