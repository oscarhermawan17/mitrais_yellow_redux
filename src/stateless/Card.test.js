import { shallow } from 'enzyme'
import React from 'react'
import Card from './Card'

describe("Card with props", () => {
    const mockModal = [{
            id:"",
            description:"",
            deadline:new Date(),
            done:true,
    }]

    // it("snapshot", () => {
    //     expect(shallow(<Card todos={mockModal} />)).toMatchSnapshot()
    // })

    it("findButton", () => {
        const wrapper = shallow(<Card todos={mockModal} />)
        wrapper.find('[htmlFor="button"]')
        expect(console.log('hasil', wrapper))
    })
})