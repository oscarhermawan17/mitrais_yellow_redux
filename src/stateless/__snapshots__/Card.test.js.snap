// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card with props snapshot 1`] = `ShallowWrapper {}`;

exports[`Card with props snapshot 2`] = `ShallowWrapper {}`;

exports[`Card with props snapshot 3`] = `ShallowWrapper {}`;
